pallets/eth-bridge/src/tx.rs

```diff
@@ -105,6 +105,12 @@ pub fn set_up_active_tx<T: Config>(req: SendRequestData) -> Result<(), Error<T>>
 }
 
 pub fn replay_send_request<T: Config>(mut tx: ActiveTransactionData<T>) -> Result<(), Error<T>> {
```

Comment:
I see this is called from `finalize_state`, and that in turn is called from `add_corroboration`. But I don't see any limit to how many times we retry, or why we want to retry if we know the transaction has already failed.

Reply:
If the sender is a malicious author, we retry until we find an honest author